searchcore/src/vespa/searchcore/bmcluster/bm_cluster.cpp

```diff
@@ -267,11 +267,17 @@ BmCluster::start_service_layers()
     start_rpc_client();
     for (const auto &node : _nodes) {
         if (node) {
-            node->wait_service_layer_slobrok(*this);
+            node->wait_service_layer_slobrok();
         }
     }
-    BmClusterController fake_controller(get_rpc_client());
-    fake_controller.set_cluster_up(false);
+    BmClusterController fake_controller(get_rpc_client(), _params.get_num_nodes());
+    unsigned int node_idx = 0;
```

Comment:
Any reason for using _unsigned int_ instead of _uint32_t_ (as used in __num_nodes_ in BmClusterParams)?